### PR TITLE
Feature/check dns mx

### PR DESCRIPTION
**Pull request by Nainterceptor**
Checklist

- [x] I've added a mention in CHANGELOG.md

### CODE REVIEW

**Comment by Nainterceptor** [comment]
[Voir si on peut vérifier entrées MX à la sauvegarde d'un mail (compliance intradef.gouv.fr & dev.gouv.fr ?)](https://trello.com/c/m6teKVEY/329-voir-si-on-peut-v%C3%A9rifier-entr%C3%A9es-mx-%C3%A0-la-sauvegarde-dun-mail-compliance-intradefgouvfr-devgouvfr)

File: models/user.mjs [hunk 1]
```diff
@@ -54,8 +57,18 @@ const UserSchema = new Schema({
...
-      validator(v) {
-        return [].concat(config.get('whitelist_domains')).reduce((acc, cur) => acc || v.endsWith(cur), false);
+      async validator(v) {
+        if (![]
+          .concat(config.get('whitelist_domains'))
+          .reduce((acc, cur) => acc || v.endsWith(cur), false)) {
+          return false;
+        }
+        const domain = v.split('@').pop();
+        if (domain === 'localhost') {
```
Comment (Scttpr) [inline models/user.mjs]: localhost ? is it a something hard coded that will be changed in prod ?

Comment (Nainterceptor) [reply models/user.mjs]: Just to prevent failure of unit tests which are playing with @localhost addresses

Comment (Nainterceptor) [reply models/user.mjs]: And we can considere that localhost is trusted because email will never be emited

**Pull request merged**